parser/src/error.rs

```diff
@@ -47,7 +47,7 @@ pub enum ParserErrorKind {
 #[derive(Debug, Clone, PartialEq)]
 pub struct ParserErrorList {
   pub kind: ParserErrorKind,
-  pub index: Range,
+  pub range: Range,
 }
```

Comment:
This whole file wouldn't be needed with a generic error handling solution like proposed above.

Reply:
Resolved by 536acff5dde53dfc5fe80af859009e84b9ab5156